packages/migrate/src/cli.ts

```diff
@@ -1,22 +1,125 @@
 
+interface Options {
+  path: string;
+  tspVersion: string;
+}
+
 async function main() {
-  const files = await findTypeSpecFiles(process.cwd());
-  const result = await migrateTypeSpecFiles(files, migrateModelToScalar);
+  console.log(`TypeSpec migration tool v${MANIFEST.version}\n`);
 
-  if (result.fileChanged.length === 0) {
-    console.log("No typespec files migrated, no change detected.");
-  } else {
-    console.log(`Updated ${result.fileChanged.length} typespec files:`);
-    for (const file of result.fileChanged) {
-      console.log(` - ${file}`);
+  const cliOptions: Options = await yargs(process.argv.slice(2))
+    .option("path", {
+      alias: "p",
+      describe: "Path to the input file. Default to current folder.",
+      type: "string",
+      default: process.cwd(),
+    })
+    .option("tspVersion", {
+      alias: "t",
+      describe: "Specifies the version of TypeSpec doc. Default will be loading from package.json",
+      type: "string",
+      default: "",
+    })
+    .help().argv;
+
+  const PackageJsonFile = "package.json";
+  if (cliOptions.tspVersion.length === 0) {
+    // Locate current package.json
+    const pkgFile = resolvePath(cliOptions.path, PackageJsonFile);
+    const packageJson: NodePackage = JSON.parse(await readFile(pkgFile, "utf-8"));
+
+    // Locate current compiler version
+    const CadlCompiler = "@cadl-lang/compiler";
+    const TypeSpecCompiler = "@typespec/compiler";
+    if (
+      packageJson?.devDependencies !== undefined &&
+      packageJson?.devDependencies[CadlCompiler] !== undefined
+    ) {
+      cliOptions.tspVersion = packageJson.devDependencies[CadlCompiler];
+    } else if (
+      packageJson?.devDependencies !== undefined &&
+      packageJson?.devDependencies[TypeSpecCompiler] !== undefined
+    ) {
+      cliOptions.tspVersion = packageJson.devDependencies[TypeSpecCompiler];
+    } else {
+      console.error("Unable to find TypeSpec compiler version in package.json.");
+      return;
+    }
+  }
+
+  if (!fs.existsSync(cliOptions.path)) {
+    console.error(`Path not found. ${cliOptions.path}`);
+    return;
+  }
+
+  let changesMake = false;
+
+  // Iterate thru migration configuration and invoke
+  console.log(`Current Typespec version ${cliOptions.tspVersion}.`);
+  const stepKeys = Object.keys(migrationConfigurations);
+  for (const key of stepKeys) {
+    if (semver.gt(key, cliOptions.tspVersion)) {
+      console.log(
+        `Migration step found to upgrade from ${cliOptions.tspVersion} to ${key}. Migrating...`
+      );
+
+      for (const migrationStep of migrationConfigurations[key]) {
+        switch (migrationStep.kind) {
+          case MigrationKind.Content:
+            const files = await findTypeSpecFiles(cliOptions.path);
+            const result = await migrateTypeSpecFiles(files, migrationStep);
+            // If migration has been performed log status
+            if (result.fileChanged.length > 0) {
```

Comment:
I think this is pre-existing but it should have been named `filesChanged` (plural)